spec/message.abnf

```diff
@@ -52,16 +55,19 @@ unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
 
 
-; reserve sigils for private-use by implementations
-private-use    = private-start reserved-body
-private-start  = "^" / "&"
+; Reserve additional .keywords for use by future versions of this specification.
+reserved-statement = reserved-keyword [s reserved-body] 1*([s] expression)
```

Comment:
Replying to @gibson042 from https://github.com/unicode-org/message-format-wg/pull/529#discussion_r1408383628:
> > Please correct me if I misremember, but my understanding from today is that we're fine with reserving only statements that end in an expression?
> 
> That is not my recollection, and would preclude possibilities like `.strict true`. However, I'm fine with addressing it in a followup.

Could we merge this PR with this note in the ABNF, in expectation of getting a PR from @gibson042 that could update all reserved content holistically?

```suggestion
; NOTE: Rules for reserved statements and annotations
; should be considered provisional, pending a near-future PR updating them.
reserved-statement = reserved-keyword [s reserved-body] 1*([s] expression)
```

ps. I had not considered statements like `.strict true`. That does sound like something we ought to reserve space for.

Reply:
The reason statements end with `expression` is that the closing `}` of the expression is how we know the statement ended. Otherwise we're in `reserved-body`... forever. It is a hidden detail of our syntax that statements end with a close bracket.

Reply:
While that's true with the current state of this PR, I believe @gibson042 would like us to drop `.` as a valid character in `reserved-body`, which would allow determining its end by encountering it at the start of the next keyword.

My request here is for us to find a way to not block this PR on that discussion, and to have it instead separately so that the syntax of both reserved statements and annotations could be considered in one package.

Reply:
I agree with finalizing this PR. I don't think the reserved-statement discussion (were we to change something) needs to be in the way of that. It should be a separate conversation. If we want to pursue a change, it should be at least an issue if not a design document.

That said, I would not include the proposed note in this comment thread. Any changes we consider or make will be done in the near future. If we include text about it, put it in syntax.md.